CRM/Event/Form/SelfSvcTransfer.php

```diff
@@ -288,7 +295,8 @@ public static function checkRegistration($fields, $self, $contact_id, &$errors)
     if (!empty($to_event_id)) {
       foreach ($to_event_id as $id) {
         if ($id == $self->_event_id) {
-          $errors['email'] = $display_name . ts(" is already registered for this event");
+          $error_field = $fields['contact_id'] ? 'contact_id' : 'email';
```

Comment:
Yes, fixed. Actually, further testing showed this was a little funky because it was only matching on email and so if you entered an email address with a different name, it came back with an error about the name associated with that email address, when the email address was the actual error, so I've corrected that too.